web/helpers/state.helper.ts

```diff
@@ -17,3 +17,37 @@ export const sortStates = (states: IState[]) => {
     return Object.keys(STATE_GROUPS).indexOf(stateA.group) - Object.keys(STATE_GROUPS).indexOf(stateB.group);
   });
 };
+
+export const getCurrentStateSequence = (groupSates: IState[], destinationData: any, edge: string | undefined) => {
+  const currentSequence = 65535;
+  if (!edge) return currentSequence;
+  const currentStateIndex = groupSates.findIndex((state) => state.id === destinationData.id);
+
+  if (destinationData.isFirstElement && destinationData.isLastElement) {
```

Comment:
Too many if conditions doing almost the same thing, break it down into sub methods and reuse them